spec-tests/src/test/java/org/candlepin/spec/CloudRegistrationSpecTest.java

```diff
@@ -431,6 +437,159 @@ public void shouldNotExportCertificatesWithAnonymousTokenForUnknownAnonCloudCons
             .exportCertificates(StringUtil.random("unknown-"), null));
     }
 
+    @Test
+    public void shouldCreateConsumerWhenUsingStandardToken() throws JsonProcessingException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        adminClient.hosted().createOwner(owner);
+        ProductDTO prod = adminClient.ownerProducts().createProductByOwner(owner.getKey(), Products.random());
+        adminClient.hosted().createProduct(prod);
+        adminClient.owners().createPool(owner.getKey(), Pools.random(prod));
+        adminClient.hosted().createSubscription(Subscriptions.random(owner, prod));
+
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+
+        associateProductIdToCloudOffer(adminClient, offerId, List.of(prod.getId()));
+        associateOwnerToCloudAccount(adminClient, accountId, owner.getKey());
+
+        String metadata = buildMetadataJson(ApiClient.MAPPER, accountId, instanceId, offerId);
+
+        CloudAuthenticationResultDTO result = adminClient.cloudAuthorization()
+            .cloudAuthorizeV2(accountId, instanceId, offerId, "test-type", "");
+
+        assertTokenType(ApiClient.MAPPER, result.getToken(), STANDARD_TOKEN_TYPE);
+
+        ConsumerDTO consumer = ApiClients.bearerToken(result.getToken()).consumers()
+            .createConsumer(Consumers.random(owner));
+
+        assertNotNull(consumer);
+    }
+
+    @Test
+    public void shouldVerifyAnonymousTokenRegistration() throws JsonProcessingException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO ownerDTO = Owners.random();
+        ProductDTO productDTO = Products.random();
+        adminClient.hosted().createOwner(ownerDTO);
+
+        adminClient.hosted().createProduct(productDTO);
+        adminClient.hosted().createSubscription(Subscriptions.random(ownerDTO, productDTO));
+
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+
+        associateProductIdToCloudOffer(adminClient, offerId, List.of(productDTO.getId()));
+        associateOwnerToCloudAccount(adminClient, accountId, ownerDTO.getKey());
+
+        String metadata = buildMetadataJson(ApiClient.MAPPER, accountId, instanceId, offerId);
+
+        CloudAuthenticationResultDTO result = adminClient.cloudAuthorization()
+            .cloudAuthorizeV2(accountId, instanceId, offerId, "test-type", "");
+
+        assertTokenType(ApiClient.MAPPER, result.getToken(), ANON_TOKEN_TYPE);
+
+        // No product and no owner exist in Candlepin
```

Comment:
@bakajstep In addition to the 2 other checks for the earlier states (Org does not exist upstream, and org exists upstream but is not entitled upstream), can you also check that the response (429) has the correct retry-after value for each of the checks (the specific seconds defined for each)?

Reply:
This still needs the the other 2 checks, for keep-alive 300 and keep-alive 180

Reply:
done